### PR TITLE
Cleanup Unneeded Modules in Setup

### PR DESCRIPTION
**Pull request by WillAyd**
Lot's of cruft in here

### CODE REVIEW

File: tabpy-tools/tabpy_tools/rest.py [hunk 1]
```diff
@@ -239,7 +238,7 @@ def __init__(self, name, bases, dict):
                 self.__rest__.add(k)
 
 
-class RESTObject(with_metaclass(_RESTMetaclass, _MutableMapping)):
+class RESTObject(_MutableMapping, metaclass=_RESTMetaclass):
```
Comment (WillAyd) [inline tabpy-tools/tabpy_tools/rest.py]: The intention here is very difficult to reason about, but that's an artifact of existing code so doing minimum here to remove third party dependency

File: tabpy-server/server_tests/test_config.py [hunk 2]
```diff
@@ -3,10 +3,7 @@
 import logging
 import tabpy_server
 
-try:
-    from unittest.mock import patch, call
-except ImportError:
-    from mock import patch, call
+from unittest.mock import patch, call
```
Comment (WillAyd) [inline tabpy-server/server_tests/test_config.py]: Given bump to Py3.6 don't need to try...except here. There's lots of other compat code that can be removed in subsequent PRs

**Comment by 0golovatyi** [comment]
@WillAyd we actually had to restore most of the deleted dependencies - even they are supposed to go with Python we immediately discovered Linux flavors where it is not the case :(

**Comment by WillAyd** [comment]
Not sure I follow - do you have an example of that? 

Most of these are Py2 packages so they'd essentially provide noops

**Comment by 0golovatyi** [comment]
@johng42 Could you provide examples for Will's question?

**Pull request merged**